motionplan/check.go

```diff
@@ -130,88 +115,105 @@ func checkPlanRelative(
+
+	// determine plan's starting pose
+	planStartPoseWorld, err := toWorld(zeroPosePIF, plan.Trajectory()[0])
 	if err != nil {
 		return err
 	}
 
-	currentPoseInWorld, err := toWorld(currentPoseIF, currentInputs)
+	// determine plan's ending pose
+	planEndPoseWorld, err := toWorld(zeroPosePIF, plan.Trajectory()[len(plan.Path())-1])
 	if err != nil {
 		return err
 	}
 
-	arcInputs, ok := plan.Trajectory()[wayPointIdx][checkFrame.Name()]
-	if !ok {
-		return errors.New("given checkFrame had no inputs in trajectory map at current index")
+	// setup the planOpts. Poses should be in world frame. This allows us to know e.g. which obstacles may ephemerally collide.
+	if sfPlanner.planOpts, err = sfPlanner.plannerSetupFromMoveRequest(
+		planStartPoseWorld.Pose(),
+		planEndPoseWorld.Pose(),
+		plan.Trajectory()[0],
+		worldState,
+		nil,
+		nil, // no pb.Constraints
+		nil, // no plannOpts
+	); err != nil {
+		return err
 	}
-	fullArcPose, err := checkFrame.Transform(arcInputs)
+	// change from 60mm to 30mm so we have finer interpolation along segments
+	sfPlanner.planOpts.Resolution = relativePlanOptsResolution
+
+	currentInputs := executionState.CurrentInputs()
+	wayPointIdx := executionState.Index()
+	sf := sfPlanner.frame
+
+	// construct first segment's start configuration
+	// get checkFrame's currentInputs
+	// *currently* it is guaranteed that a relative frame will constitute 100% of a solver frame's dof
+	checkFrameCurrentInputs, err := sf.mapToSlice(currentInputs)
 	if err != nil {
 		return err
 	}
 
-	// Relative current inputs will give us the arc the base has executed. Calculating that transform and subtracting it from the
-	// arc end position (that is, the same-index node in plan.Path()) gives us our expected location.
-	frameCurrentInputs, ok := currentInputs[checkFrame.Name()]
-	if !ok {
-		return errors.New("given checkFrame had no inputs in CurrentInputs map")
-	}
-	poseThroughArc, err := checkFrame.Transform(frameCurrentInputs)
+	// construct first segment's end configuration
+	arcEndInputs, err := sf.mapToSlice(plan.Trajectory()[wayPointIdx])
 	if err != nil {
 		return err
 	}
-	remainingArcPose := spatialmath.PoseBetween(poseThroughArc, fullArcPose)
-	expectedCurrentPose := spatialmath.PoseBetweenInverse(remainingArcPose, expectedArcEndInWorld.Pose())
-	errorState := spatialmath.PoseBetween(expectedCurrentPose, currentPoseInWorld.Pose())
 
-	planStartPiF, ok := plan.Path()[0][checkFrame.Name()]
+	// construct first segment's startPosition
+	currentPoses := executionState.CurrentPoses()
+	if currentPoses == nil {
+		return errors.New("executionState had nil return from CurrentPoses")
+	}
+
+	currentPoseIF, ok := currentPoses[checkFrame.Name()]
 	if !ok {
-		return errors.New("check frame given not in plan Path map")
+		return errors.New("checkFrame not found in current pose map")
 	}
-	planStartPoseWorld, err := toWorld(planStartPiF, startingInputs)
+	err = validateRelPiF(currentPoseIF)
```

Comment:
This should no longer need to be checked, right? I think `validateRelPiF` can just be deleted

Reply:
I think that that `validateRelPiF` should stay for this one instance.
My reasoning is that we read `currentPoseIF` from `currentPoses` which comes from `executionState.CurrentPoses()`.
I think we should not make assumptions that `currentPoseIF` already has its parent as `referenceframe.World` and should ensure that though `validateRelPiF`

Reply:
But we should use `toWorld` then, no?
`validateRelPiF` was specifically to make sure that the pose given of a relative frame was anchored on something absolute so that the relative transform could be applied; now that the wrapper frame is absolute, we can determine the pose even if the pose is given in a downstream frame.

Reply:
> But we should use toWorld then, no?

Yup, we use `toWorld` right after we call `validateRelPiF`.

>validateRelPiF was specifically to make sure that the pose given of a relative frame was anchored on something absolute so that the relative transform could be applied

^^ this was my mental error that I made - thank you for pointing this out and giving me the opportunity to make this code even cleaner.

I understand your comment now. I will remove `validateRelPiF` 🗑️